shim/src/main.rs

```diff
@@ -15,50 +19,96 @@ use std::{
+#[derive(Parser, Debug, Serialize, Default, PartialEq)]
+#[clap(author, version, about = "Turbocharge your monorepo", long_about = None, ignore_errors = true, disable_help_subcommand = true)]
 struct Args {
-    /// Current working directory
-    #[clap(long, value_parser)]
+    /// Override the endpoint for API calls
+    #[clap(long, global = true, value_parser)]
+    api: Option<String>,
+    /// Force color usage in the terminal
+    #[clap(long, global = true, value_parser)]
+    color: bool,
+    /// Specify a file to save a cpu profile
+    #[clap(long, global = true, value_parser)]
+    cpuprofile: Option<String>,
+    /// The directory in which to run turbo
+    #[clap(long, global = true, value_parser)]
     cwd: Option<String>,
+    /// Specify a file to save a pprof heap profile
+    #[clap(long, global = true, value_parser)]
+    heap: Option<String>,
+    /// Override the login endpoint
+    #[clap(long, global = true, value_parser)]
+    login: Option<String>,
+    /// Suppress color usage in the terminal
+    #[clap(long, global = true, value_parser)]
+    no_color: bool,
+    /// When enabled, turbo will precede HTTP requests with an OPTIONS request for authorization
+    #[clap(long, global = true, value_parser)]
+    preflight: bool,
+    /// Set the team slug for API calls
+    #[clap(long, global = true, value_parser)]
+    team: Option<String>,
+    /// Set the auth token for API calls
+    #[clap(long, global = true, value_parser)]
+    token: Option<String>,
+    /// Specify a file to save a pprof trace
+    #[clap(long, global = true, value_parser)]
+    trace: Option<String>,
+    /// verbosity
+    #[clap(short, long, global = true, value_parser)]
+    verbosity: Option<u8>,
     #[clap(subcommand)]
     command: Option<Command>,
-    task: Option<String>,
+    tasks: Vec<String>,
 }
 
+static TURBO_JSON: &str = "turbo.json";
+
 /// Defines the subcommands for CLI. NOTE: If we change the commands in Go,
 /// we must change these as well to avoid accidentally passing the --single-package
 /// flag into non-build commands.
-#[derive(Subcommand, Debug)]
+#[derive(Subcommand, Debug, Serialize, PartialEq)]
 enum Command {
+    /// Get the path to the Turbo binary
     Bin,
+    /// Generate the autocompletion script for the specified shell
     Completion,
+    /// Runs the Turborepo background daemon
     Daemon,
+    /// Help about any command
     Help,
+    /// Link your local directory to a Vercel organization and enable remote caching.
     Link,
+    /// Login to your Vercel account
     Login,
+    /// Logout to your Vercel account
     Logout,
+    /// Prepare a subset of your monorepo.
     Prune,
-    Unlink,
+    /// Run tasks across projects in your monorepo
     Run { tasks: Vec<String> },
+    /// Unlink the current directory from your Vercel organization and disable Remote Caching
+    Unlink,
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone, Serialize)]
```

Comment:
Do we need them yet? Seems like we should wait until they're necessary, no?

Reply:
This is a total bikeshed so no need to change anything.

I tend to over impl standard traits especially in a codebase where Rust is a newer language. When I try to put something in a `HashSet` or use `==` and it fails due to a missing implementation, I generally assume that either the type can't/shouldn't implement the trait or there's something about the type that makes the derived impl incorrect.

Less applicable here since this is a private struct: [C-COMMON-TRAITS](https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits)